src/main/java/spring/concept/controllers/UserController.java

```diff
@@ -0,0 +1,46 @@
+package spring.concept.controllers;
+
+import java.util.List;
+import java.util.stream.Collectors;
+import org.springframework.context.annotation.AnnotationConfigApplicationContext;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+import spring.concept.config.AppConfig;
+import spring.concept.dto.UserResponseDto;
+import spring.concept.model.User;
+import spring.concept.service.UserService;
+
+@RestController
+@RequestMapping("/user")
+public class UserController {
+    private AnnotationConfigApplicationContext context =
+            new AnnotationConfigApplicationContext(AppConfig.class);
+    private UserService userService = context.getBean(UserService.class);
+
+    @GetMapping("/inject")
+    public String injectUsers() {
+        userService.add(new User("Andy", "a@gmail.com", "1234", "Andy R"));
+        userService.add(new User("Mandy", "m@gmail.com", "asdf1234", "Mandy B"));
+        userService.add(new User("Gareth", "ga@gmail.com", "wert4321", "Gareth J"));
+        userService.add(new User("Gwyn", "gw@gmail.com", "98oiuy", "Gwyn As"));
+        return "Users were injected.";
+    }
+
+    @GetMapping("/{id}")
+    public UserResponseDto getUser(@PathVariable Long id) {
+        return mapUserToDto(userService.getById(id));
+    }
+
+    @GetMapping()
+    public List<UserResponseDto> getAll() {
+        return userService.listUsers().stream()
+                .map(this::mapUserToDto)
+                .collect(Collectors.toList());
+    }
+
+    private UserResponseDto mapUserToDto(User user) {
+        return new UserResponseDto(user.getName(), user.getEmail());
```

Comment:
let's user setters here to explicitly show what we're setting to dto

Reply:
> let's user setters here to explicitly show what we're setting to dto

you can leave this impl, as long as you don't pass model of user to constructor, my mistake

Reply:
Sure. But - What is the difference? What is the impact of my first solution?

Reply:
> Sure. But - What is the difference? What is the impact of my first solution?

as I mentioned you can leave your first solution, that won't make difference. I just made mistake and thought that user model was passed to constructor and that was not that good, since in this case it's not clear what fields are used etc.
So in conclusion both impl are fine and it's up to you